docs/DeveloperGuide.md

```diff
@@ -36,3 +33,115 @@
+
+**Step 2.** User input is read (multiple lines) until a blank line is found.
+Then, `reviewText` is updated with the text input.
+
+**Step 3.** Next, user is prompted for a numerical input (with input validation).
+This value is updated into `reviewStars`.
+
+**Step 4.** Finally, `Scanner` is closed and `Review` is done setting.
+
+# Movie List Feature
+**API:** [Movie.java](https://github.com/AY2223S2-CS2113-W12-4/tp/blob/master/src/main/java/MovieList.java)
+
+Here's a partial class diagram of the `MovieList` component:
+
+![](/PUMLFiles/MovieList/movielist1.png)
+
+The `MovieList` component has two inherited classes `ToWatchList` and `WatchedList`. It contains a public method `movieList` for creating an empty movie list,
+which also has a overloaded constructor for creating a movie list with an existing list.
+In addition, `MovieList` contains a public method `add`, which will search for relevant movies in the database
+and help user to add a new movie to the movie list. It also contains a public method `filter`, 
+which will filter the movies with certain genre and display it to the user. The public method `remove`, will remove a movie from the list, 
+and this will be implemented soon. Other methods such as `getMovie()` and `toString()` are also included in `MovieList`. 
+
+
+How the `MovieList.add(String inputTitle)` component works:
+
+![](/PUMLFiles/MovieList/movielist2.png)
+
+**Step 1.** `add(String inputTitle)` is called, which continues to call `find()` in `MovieDatabase` to search for the relevant movies.
+It will then display at most five movies that matches the `inputTitle`.
+
+**Step 2.** Next, user is prompted for a numerical input (with input validation).
+This value indicates the movie index chosen by the user from the list.
+
+**Step 3.** Next, `createMovie` is called to create a new `Movie` object according to the index and add the movie to the movie list.
+
+**Step 4.** The `add(String inputTitle)` will show feedback message to the user.
+
+**Step 5.** The `add(String inputTitle)` has finished everything and returned.
+
+How the `MovieList.remove()` component works:
+
+![](/PUMLFiles/MovieList/movielist3.png)
+
+**Step 1.** `remove` command is entered, which continues to call `showListMessage()` in `Ui` to 
+show the user the movies in the list.
+It will then display all movies from `watched list` or `to-watch list`, allowing the user to easily pick the movie for deletion.
+
+**Step 2.** Next, user is prompted for a numerical input (with input validation).
+This value indicates the movie index chosen by the user from the list.
+
+**Step 3.** Next, `remove()` is called to remove a  `Movie` object according to the index and delete the movie to the movie list.
+
+**Step 4.** The function will call `showDeleteMessage` from `Ui`, which will show feedback message to the user.
+
+**Step 5.** The `remove` command has finished everything and ended.
+
+How the `MovieList seedetail` component works:
```

Comment:
This function is not following the naming convention for functions, maybe can call it seeDetail instead